apis/v1beta1/gatewayclass_types.go

```diff
@@ -183,6 +183,21 @@ const (
 
 // GatewayClassStatus is the current status for the GatewayClass.
 type GatewayClassStatus struct {
+	// Routabilities specifies a list of supported routabilities offered by
+	// the GatewayClass. The first entry in this list will be the default
+	// routability used when Gateways of this class are created.
```

Comment:
This feels error prone, you'll need to make sure the first entry is never overwritten. Why not add a defaultRoutability field to a new type that contains the default and []GatewayRoutability?

Reply:
This is in the status block of the GatewayClass which is reconciled by implementations. What is error prone here?

> Why not add a defaultRoutability field to a new type that contains the default and []GatewayRoutability?

This seems more complex for clients to consume - how do I get the entire list of routoabilities? do I need to concat the default with the list? then do I need to dedupe entries the list?

Reply:
This comment is applicable to the PR here: https://github.com/kubernetes-sigs/gateway-api/pull/2157

If this is still a concern please comment there